### PR TITLE
OGC-189: Implement Lab unit enable/disable

### PR DESCRIPTION
**Pull request by reagan-meant**
Core problem: deactivating a lab unit was purely cosmetic — it hid the unit from one admin list and nothing else. Tests still routed to it, orders still landed in it, results still flowed through it.

Lab unit deactivation now works

1. A deactivated lab unit stops accepting new orders — enforced server-side across manual entry, reflexes, and analyzers
2. The rule is derived (test.active && labUnit.isActive), never written to the tests, so reactivating restores everything with no cleanup step
3. Switching a unit off now shows an impact summary with live counts, three choices (keep assignments / deactivate all / reassign), and requires typing DEACTIVATE
4. Reflexes no longer fire into a disabled unit, and the block is logged loudly since nobody is watching when an automatic test fails to fire

Work already in progress is never stranded

1. In-flight and completed analyses stay visible and completable after their lab unit is switched off — across results entry, validation, workplan, reports, and history
2. Same protection when the test itself is deactivated
3. Entering a result no longer makes it vanish from the results pages and reports
4. A reflexed analysis filed in an active unit is no longer hidden because its test is configured elsewhere
5. Six separate causes, fixed at six different stages of the order → complete → view → report path

Visibility corrected in both directions

1. The test editor no longer offers deactivated lab units as new choices
2. A test already assigned to a deactivated unit keeps showing it, marked (inactive), so saving can't silently wipe the assignment
3. Worklists and report filters keep showing a deactivated unit for as long as it holds work

Smaller fixes

1. The 20-character lab unit name limit is now enforced when editing, not only when adding
2. Order entry: searching for a test that doesn't exist no longer breaks test selection permanently — the search box used to disappear along with the results, leaving no way to clear the term

# Pull Requests Requirements

- [ ] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [ ] The PR title follows conventional commit label standards.
- [ ] The changes confirm to the OpenElis Global x3
      [Styleguide and Design](https://uwdigi.atlassian.net/wiki/spaces/OG/pages/621346838/OpenELIS+Global+Style+Guide)
      documentation.
- [ ] The changes include tests or are validated by existing tests.
- [ ] I have read and agree to the Contributing
      [Guidelines](https://openelis-global.org/community/get-involved/) of this
      project.

## Screenshots

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other

[Add any additional information or notes here]

### CODE REVIEW

File: frontend/src/components/admin/labUnitManagement/sections/AssignedTestsSection.jsx [hunk 1]
```diff
@@ -465,6 +465,13 @@ function AssignedTestsSection({ labUnitId, onChange }) {
                 id: "placeholder.labUnit.tests.reassign.destination",
               })}
             />
+            {/* OGC-189 (M2, decision D4): reassign destinations are
+                DELIBERATELY unfiltered — inactive units included. Ruled
+                acceptable 2026-09-01 ("fine with moving to a switched off
+                lab"), because emptying a unit into a deactivated one is a
+                legitimate reorganisation step. This is the documented
+                exception to the chooser rule that the test editor's lab unit
+                picker follows; do not "fix" it by filtering on isActive. */}
```
Comment (mherman22) [inline frontend/src/components/admin/labUnitManagement/sections/AssignedTestsSection.jsx]:
```suggestion
            { }
```

File: frontend/src/components/admin/labUnitManagement/LabUnitManagement.jsx [hunk 2]
```diff
@@ -381,7 +394,10 @@ function LabUnitManagement({ intl }) {
         errors[nameErrorKey] = intl.formatMessage({
           id: "error.labUnit.name.required",
         });
-      } else if (view === "add" && name.length > NAME_MAX_LENGTH) {
+      } else if (name.length > NAME_MAX_LENGTH) {
+        // The cap applies on edit as well as add — it was gated behind
+        // `view === "add"`, so a longer name saved and persisted (OGC-189, QA
+        // LU-W-3).
```
Comment (mherman22) [inline frontend/src/components/admin/labUnitManagement/LabUnitManagement.jsx]:
```suggestion
      } else if (name.length > NAME_MAX_LENGTH) {
        // The cap applies on edit as well as add. It was gated behind
        // `view === "add"`, so a longer name saved and persisted).
```

File: src/main/java/org/openelisglobal/typeofsample/service/TypeOfSampleServiceImpl.java [hunk 3]
```diff
@@ -101,8 +104,12 @@ public synchronized List<Test> getActiveTestsBySampleTypeId(String sampleTypeId,
     public synchronized List<Test> getActiveTestsBySampleTypeIdAndTestUnit(String sampleType, boolean b,
             List<String> testUnitIds) {
         List<Test> testList = getActiveTestsBySampleTypeId(sampleType, b);
+        // OGC-189 (M4): also drop tests whose lab unit is switched off. This
+        // filter used to consider only the test's own active flag, so a
+        // deactivated lab unit kept taking orders (QA LU-W-11/LU-W-12) — the
+        // unit's status did not participate in orderability at all.
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/typeofsample/service/TypeOfSampleServiceImpl.java]:
```suggestion

```

File: src/main/java/org/openelisglobal/testreflex/action/util/ReflexAction.java [hunk 4]
```diff
@@ -90,6 +93,40 @@ protected void createReflexedAnalysis(Test test) {
...
+            // note on setTestSection below) — so that is the unit to test, not
+            // the reflexed test's own configured one.
+            //
+            // Gating on the reflexed test's own unit instead would get both
+            // edge cases wrong: it would block a reflex that was going to land
+            // in a perfectly active parent unit, and let one through into a
+            // deactivated parent unit.
+            //
+            // The test's own active flag still applies — an inactive test is
+            // not orderable by any route.
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/testreflex/action/util/ReflexAction.java]:
```suggestion
             // A reflex must only be filed under an active parent lab unit. The reflexed test must also be active/orderable; its own lab unit is irrelevant.
```

File: src/main/java/org/openelisglobal/test/service/TestService.java [hunk 5]
```diff
@@ -31,6 +31,9 @@ public interface TestService extends BaseObjectService<Test, String> {
 
     List<Test> getTestsByTestSectionIds(List<String> ids);
 
+    /** OGC-189: unfiltered by active status, for viewer paths. See TestDAO. */
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/test/service/TestService.java]:
```suggestion
    /** Unfiltered by active status, for viewer paths. See TestDAO. */
```

File: src/main/java/org/openelisglobal/test/service/EffectiveTestStatusServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
+package org.openelisglobal.test.service;
+
+import org.apache.commons.validator.GenericValidator;
+import org.openelisglobal.test.valueholder.Test;
+import org.openelisglobal.test.valueholder.TestSection;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/** OGC-189 (M4). See {@link EffectiveTestStatusService}. */
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/test/service/EffectiveTestStatusServiceImpl.java]:
```suggestion
/** See {@link EffectiveTestStatusService}. */
```

File: src/main/java/org/openelisglobal/systemuser/service/UserServiceImpl.java [hunk 7]
```diff
@@ -408,32 +494,52 @@ public List<IdValuePair> getAllDisplayUserTestsByLabUnit(String SystemUserId, St
...
         if (testSections != null) {
             testSections.forEach(testSection -> testUnitIds.add(testSection.getId()));
         }
 
-        List<Test> allTests = testService.getTestsByTestSectionIds(testUnitIds);
-        List<String> allTestsIds = new ArrayList<>();
-        allTests.forEach(test -> allTestsIds.add(test.getId()));
-        return results.stream().filter(result -> allTestsIds.contains(result.getTest().getId()))
-                .collect(Collectors.toList());
+        // OGC-189: judge each analysis by ITS OWN lab unit, not by the unit its
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/systemuser/service/UserServiceImpl.java]:
```suggestion
        // judge each analysis by ITS OWN lab unit, not by the unit its
```
